src/main/kotlin/org/opensearch/replication/repository/RestoreContext.kt

```diff
@@ -38,13 +41,23 @@ class RestoreContext(
         val lock = fileLocks.computeIfAbsent(fileName) { ReentrantLock() }
 
         lock.withLock {
-            val baseInput = currentFiles.computeIfAbsent(fileName) {
-                store.directory().openInput(fileName, IOContext.DEFAULT)
+            var baseInput: IndexInput? = null
+
+            withStoreReference(store) {
+                baseInput = currentFiles.computeIfAbsent(fileName) {
+                    store.directory().openInput(fileName, IOContext.DEFAULT)
+                }
             }
-            return baseInput.clone()
+
+            return checkNotNull(baseInput) { "[RestoreContext] IndexInput file must not be null" }.clone()
         }
     }
 
+    // for testing
+    internal open fun withStoreReference(store: Store, block: () -> Unit) {
+        store.performOp(block)
```

Comment:
`store.performOp` is an extension function, so it requires static function mocking, which can be a bit tricky. so I added a separate function for it.